pkg/vsphere/vm/vm.go

```diff
@@ -121,6 +121,22 @@ func (vm *VirtualMachine) FolderName(ctx context.Context) (string, error) {
 	return path.Base(u.Path), nil
 }
 
+// Folder returns a reference to the parent folder that owns the vm
+func (vm *VirtualMachine) Folder(ctx context.Context, sess *session.Session) (*object.Folder, error) {
```

Comment:
I can likely change this to use the internal session stored in the vm as well. Would simplify the function inputs further.